src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -147,5 +151,11 @@ public class TagHelperDescriptor
         /// The <see cref="TagHelperUsageDescriptor"/> that indicates how to use the tag helper at design time.
         /// </summary>
         public TagHelperUsageDescriptor UsageDescriptor { get; }
+
+        /// <summary>
+        /// The HTML element the tag helper may output.
+        /// </summary>
+        /// <remarks>Overrides the HTML IntelliSense provided at design time.</remarks>
+        public string OutputElementHint { get; set; }
```

Comment:
Hmm good question. We could:

1. Take first hint provided.
1. Concat hints (but if we do this why not allow a single `TagHelper` to provide multiple hints).
1. Error.

/cc @DamianEdwards